iree/turbine/kernel/wave/constraints.py

```diff
@@ -392,6 +393,8 @@ def count(self) -> IndexExpr:
         """
         Returns an expression for the total number of workgroups for the specific workgroup_dim.
         """
+        if self.iters:
```

Comment:
Can you explain a little when we would want something other than `ceiling(self.dim / self.tile_size)`? I thought we'd always want `grid_size/block_size`? 

Unless we are somehow splitting same data dim across mutliple thread grid dim? 🤔

Reply:
Good question. In this case, the sequence length is dynamic and so in order to determine how many workgroups to launch, we use the maximum sequence length.